caravel/assets/javascripts/explorev2/actions/exploreActions.js

```diff
@@ -48,18 +48,14 @@ export function setFilterColumnOpts(filterColumnOpts) {
 }
 
 export function resetFormData() {
-  // Clear all form data when switching datasource
+  // Clear all form data when switching datasource or viz_type
   return { type: RESET_FORM_DATA };
```

Comment:
right, so wouldn't the data that is returned from fetching the datasource meta data be set on the state, in effect 'resetting' the form data?